core/cthreads.h

```diff
@@ -0,0 +1,91 @@
+#ifndef CTHREADS_H
+#define CTHREADS_H
+
+struct cthreads_args {
+  void *(*func)(void *data);
+  void *data;
+};
+
+#if __unix__
+#include <pthread.h>
+#elif _WIN32
+#include <windows.h>
+#endif
+
+struct cthreads_thread {
+    #ifdef __unix__
+    pthread_t pThread;
+    #elif _WIN32
+    HANDLE wThread;
+    #endif
+};
```

Comment:
Okay, so, I compiled this on OS X and it kinda... didn't work. But, somehow, it did work. While I do like your approach, UNIX systems do not always define `__unix__` for some dumb reason. So, you'd probably want to use the `osname` macros and see if `OS_UNIX` is defined. Here's what you get on OS X:
```
../core/cthreads.h:15:1: warning: empty struct is a GNU extension [-Wgnu-empty-struct]
struct cthreads_thread {
```
ad infinitum. 

We should probably update the `osname` header with the latest from https://github.com/c-ware/cware-utils/blob/trunk/osname.h, too. Note that this still compiles fine, but, it does not present a clean solution to have those few missing struct members.